File: javascript/socketd/src/socketd/transport/core/Listener.ts
import type {Session} from "./Session";
import type {Message} from "./Message";

/**
 * 监听器
 *
 * @author noear
 * @since 2.0
 */
export interface Listener {
    /**
     * 打开时
     *
     * @param session 会话
     */
    onOpen(session: Session);

    /**
     * 收到消息时
     *
     * @param session 会话
     * @param message 消息
     */
    onMessage(session: Session, message: Message);

    /**
     * 收到答复时
     *
     * @param session 会话
     * @param message 消息
     */
    onReply(session: Session, message: Message);

    /**
     * 发送消息时
     *
     * @param session 会话
     * @param message 消息
     */
    onSend(session: Session, message: Message);

    /**
     * 关闭时
     *
     * @param session 会话
     */
    onClose(session: Session);

    /**
     * 出错时
     *
     * @param session 会话
     * @param error   错误信息
     */
    onError(session: Session, error: any);
}

File: python/socketd/socketd/transport/core/Listener.py
import abc

from socketd.transport.core.Message import Message
from socketd.transport.core.Session import Session


class Listener(abc.ABC):
    # 打开时
    @abc.abstractmethod
    async def on_open(self, session: Session):
        pass

    # 收到消息时
    @abc.abstractmethod
    async def on_message(self, session: Session, message: Message):
        pass

    # 收到答复时
    async def on_reply(self, session: Session, message: Message):
        pass

    # 发送消息时
    async def on_send(self, session: Session, message: Message):
        pass

    # 关闭时
    @abc.abstractmethod
    async def on_close(self, session: Session):
        pass

    # 出错时
    @abc.abstractmethod
    async def on_error(self, session: Session, error):
        pass
